src/bust.py

```diff
@@ -79,10 +77,13 @@ def finished(self) -> bool:
 
     async def stop(self) -> None:
         """Stop playing music."""
-        self.play_next_cancelled = True
-        if self.play_next_task:
-            self.play_next_task.cancel()
-        await self.finish(say_goodbye=False)
+        self.bust_stopped = True
+        if self.play_song_task:
```

Comment:
It looks like we could technically trigger this. When calling `bc.play()` there is a point after the voice client connects where Busty awaits to set herself as speaking in a stage channel (the line `await bot_member.edit(suppress=False)`), at which point control flow could switch to handling a stop command which would pass the `bc.is_active()` check in `main.py` and get to this line. Though admittedly I wasn't really thinking of this when writing, I did it so it's easier to check types and "just in case" since it's a cheap and readable future-proofing check.

Reply:
Heh, yeah fair. It doesn't hurt to add either.